js/style/style.js

```diff
@@ -420,7 +399,7 @@ Style.prototype = util.inherit(Evented, {
         }
         this._validateLayer(layer);
 
-        layer.on('error', this._forwardLayerEvent);
+        layer.setEventedParent(this, {layer: {id: layer.id}});
```

Comment:
Is it necessary to have nested `{layer: {id: ...}` data for some compatibility reasons, or could we just have `{layerId: ...}`?